orttraining/orttraining/training_api/include/onnxruntime_training_c_api.h

```diff
@@ -296,6 +296,23 @@ struct OrtTrainingApi {
   *
   */
   ORT_CLASS_RELEASE(CheckpointState);
+
+  /** \brief Export a model that can be used for inferencing.
+  *
+  * If the training session was provided with an eval model, the training session can generate
+  * an inference model if it knows the inference graph outputs. The inference graph outputs are expected
+  * to be set in the metadata_props field at the time of eval model creation. This happens by default
+  * using onnxruntime offline tooling.
+  * The exported model is saved at the path provided and can be used for inferencing with InferenceSession.
+  *
+  * \param[in] sess The training session.
+  * \param[in] inference_model_path Path where the inference model should be serialized to.
+  *
+  * \snippet{doc} snippets.dox OrtStatus Return Value
+  *
+  */
+  ORT_API2_STATUS(ExportModelForInferencing, _Inout_ OrtTrainingSession* sess,
```

Comment:
can we add model outputs as input params for this api? If they are available, then no need to use metadata for determining the model outputs

Reply:
This can be done with few changes.

Would you expect the user have different inference model than the one they started out with (during offline tooling)? Would we need this API if we can already decipher the inference model from the eval model without user intervention?

Reply:
Actually I prefer this over the metadata option. Metadata option is more error prone... plus there is a possibility that the user may edit the model or some optimization may make changes...

Reply:
The metadata option should not be error prone considering that the offline tooling only allows the eval model to be a superset of the inference model.
Since the training session requires an unmodified eval model, it would not be optimized at all (and we load the model separately from inference session, so the inference session optimizations won't impact the eval model either).

I think it is safer than the user providing the graph inputs considering they need to know the string name for such inputs without making any string related errors (typos...).

Having said that, if we feel it is still needed, I'll add the change.

Reply:
I changed the api to accept graph output names as string inputs. I am ok with this change, but I still feel having the metadata filed is a cleaner solution (which should never yield any errors) which does not require any user intervention on generating the inference graph. If we want, maybe we can go back to this solution in the future.

Reply:
Just make sure I understand it clearly, the inference graph output list is not an optional param, am I right? I personally don't like options as users, assuming I need read more docs when metadata works and when explict graph output list works.

Reply:
I removed the metadata approach. Now, the user is expected to pass in the expected inference graph output. The api will perform graph modification based on this vector of output names.